packages/gasket-plugin-nextjs/lib/webpack-config.js

```diff
@@ -0,0 +1,57 @@
+const isGasketCore = /@gasket[/\\](core)/;
+
+function validateNoGasketCore(ctx, callback) {
+  if (isGasketCore.test(ctx.request)) {
+    return callback(new Error('@gasket/core should not be used in browser code.'));
+  }
+  return callback();
+}
+
+/**
+ * Externalize @gasket/core in the server build.
+ * We do this to enable GASKET_ENV to be picked up and passed to the plugins.
```

Comment:
I just wonder if there's a way we can avoid using an environment variable altogether except in the case of initialization.

Reply:
As we move away from the CLI-driven approach, using this env var will be more necessary.